### PR TITLE
Minor revisions to custom layer.

### PR DESCRIPTION
**Pull request by vishaalkapoor**
Minor clarifications, link corrections. Content looks complete.

### CODE REVIEW

**Comment by vishaalkapoor** [comment]
@astonzhang would you mind taking a look, thanks!

File: python/guide/module/gluon/custom-layer.md [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
 One of the reasons for the success of deep learning can be found in the wide range of layers that can be used in a deep network. This allows for a tremendous degree of customization and adaptation. For instance, scientists have invented layers for images, text, pooling, loops, dynamic programming, even for computer programs. Sooner or later you will encounter a layer that doesn't exist yet in Gluon, or even better, you will eventually invent a new layer that works well for your problem at hand. This is when it's time to build a custom layer. This section shows you how.
 
+Defining a custom layer (aka Block) is as easy as subclassing `nn.Block` or `nn.HybridBlock` and implementing `forward`. To take advantage of the performance gains with `nn.HybridBlock` see see the section on [Hybridization](hybridize.html). We will discuss making custom layers using `nn.Block`, below.
```
Comment (astonzhang) [inline python/guide/module/gluon/custom-layer.md]: see see -> see

Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: thanks. updated

**Comment by vishaalkapoor** [comment]
Seems https://conda.anaconda.org/pytorch/noarch/repodata.json is currently timing out (504) so the CI is failing.

File: python/guide/module/gluon/custom-layer.md [hunk 2]
```diff
@@ -83,7 +85,7 @@ dense.initialize()
 dense(nd.random.uniform(shape=(2, 5)))
 ```
 
-We can also construct models using custom layers. Once we have that we can use it just like the built-in dense layer. The only exception is that in our case size inference is not automagic. Please consult the [MXNet documentation](http://www.mxnet.io) for details on how to do this.
+We can also construct models using custom layers. Once we have that we can use it just like the built-in dense layer. The only exception is that in our case size inference is not automatic.
```
Comment (thomelane) [inline python/guide/module/gluon/custom-layer.md]: size inference is not automatic. would explain why not. also size -> shape

File: python/guide/module/gluon/custom-layer.md [hunk 3]
```diff
@@ -4,9 +4,11 @@
 
 One of the reasons for the success of deep learning can be found in the wide range of layers that can be used in a deep network. This allows for a tremendous degree of customization and adaptation. For instance, scientists have invented layers for images, text, pooling, loops, dynamic programming, even for computer programs. Sooner or later you will encounter a layer that doesn't exist yet in Gluon, or even better, you will eventually invent a new layer that works well for your problem at hand. This is when it's time to build a custom layer. This section shows you how.
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: added link.

[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: http://ceur-ws.org/Vol-1583/CoCoNIPS_2015_paper_1.pdf

File: python/guide/module/gluon/custom-layer.md [hunk 4]
```diff
@@ -4,9 +4,11 @@
...
 
+Defining a custom layer (aka `Block`) is as easy as subclassing `nn.Block` or `nn.HybridBlock` and implementing `forward`. To take advantage of the performance gains with `nn.HybridBlock` see the section on [Hybridization](hybridize.html). We will discuss making custom layers using `nn.Block`, below.
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: Tightened up.

File: python/guide/module/gluon/custom-layer.md [hunk 5]
```diff
@@ -4,9 +4,11 @@
...
+
 ## Layers without Parameters
 
-Since this is slightly intricate, we start with a custom layer (aka Block) that doesn't have any inherent parameters. Our first step is very similar to when we [introduced blocks](model-construction.md) previously. The following `CenteredLayer` class constructs a layer that subtracts the mean from the input. We build it by inheriting from the Block class and implementing the `forward` method.
+Since this is slightly intricate, we start with a custom layer that doesn't have any inherent parameters. Our first step is very similar to when we [introduced blocks](nn.md) previously. The following `CenteredLayer` class constructs a layer that subtracts the mean from the input. We build it by inheriting from the `Block` class and overriding the `forward`  and `__init__` methods.
 
 ```{.python .input  n=1}
 from mxnet import gluon, nd
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: added print statements where appropriate as discussed offline, made other changes.

File: python/guide/module/gluon/custom-layer.md [hunk 6]
```diff
@@ -46,7 +48,7 @@ y.mean().asscalar()
 
 Now that we know how to define layers in principle, let's define layers with parameters. These can be adjusted through training. In order to simplify things for an avid deep learning researcher the `Parameter` class and the `ParameterDict` dictionary provide some basic housekeeping functionality. In particular, they govern access, initialization, sharing, saving and loading model parameters. For instance, this way we don't need to write custom serialization routines for each new custom layer.
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: Done.

There isn't an API link for ParameterDict. Added for Parameter.

File: python/guide/module/gluon/custom-layer.md [hunk 7]
```diff
@@ -46,7 +48,7 @@ y.mean().asscalar()
...
 
-For instance, we can use the member variable `params` of the `ParameterDict` type that comes with the Block class. It is a dictionary that maps string type parameter names to model parameters in the `Parameter` type.  We can create a `Parameter` instance from `ParameterDict` via the `get` function.
+For instance, we can use the member variable `params` of the `ParameterDict` type that comes with the `Block` class. It is a dictionary that maps string type parameter names to model parameters in the `Parameter` type.  We can create a `Parameter` instance from `ParameterDict` via the `get` function which attempts to retrieve a parameter, or create it if not found.
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: done

File: python/guide/module/gluon/custom-layer.md [hunk 8]
```diff
@@ -46,7 +48,7 @@ y.mean().asscalar()
...
 
 ```{.python .input  n=7}
 params = gluon.ParameterDict()
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: Done (hybrid_forward mentioned in second paragraph)

File: python/guide/module/gluon/custom-layer.md [hunk 9]
```diff
@@ -83,7 +85,7 @@ dense.initialize()
 dense(nd.random.uniform(shape=(2, 5)))
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: print(dense)

File: python/guide/module/gluon/custom-layer.md (hunk 2 above)
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: done

**Comment by vishaalkapoor** [comment]
@thomelane  Please take a look

File: python/guide/module/gluon/custom-layer.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply python/guide/module/gluon/custom-layer.md]: edit: 
reworded

**Comment by thomelane** [comment]
LGTM, just a conflict to resolve, then can merge.

**Pull request merged**